### ISSUE TITLE
Impossible to place the cursor after a mark (bold, code, etc.)

### ISSUE THREAD

**Issue by rfgamaral** [ueberdosis/tiptap#2572]
### What’s the bug you are facing?

When a mark, such as block, code, etc, is at the end of a paragraph, it's impossible to place the cursor after said mark. Here's a demo with the [official Code](https://tiptap.dev/api/marks/code) example:

![PowerToys_ZGqX587oSI](https://user-images.githubusercontent.com/96476/155422397-49745eab-48cb-42e8-864d-74ce8df8a4a9.gif)

### How can we reproduce the bug on our side?

See the demo above 👆

### Can you provide a CodeSandbox?

n/a

### What did you expect to happen?

One should always be able to type anything after a mark, even if it's the last mark on a paragraph.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [X] Yes, I’m a sponsor. 💖 _(on behalf of Doist)_

**Comment by sjdemartini** [comment]
I'm also encountering this problem. This seems related to these other two issues https://github.com/ueberdosis/tiptap/issues/514, https://github.com/ueberdosis/tiptap/issues/2571 (the first of which is a longstanding one).

**Comment by rfgamaral** [comment]
I'm assuming this is the default behavior, and that one can "exit" the mark by using a keyboard shortcut (or the toolbar buttons, if available). However, I don't think the `Link` extension should work like that, and I've personally disabled this for us: https://tiptap.dev/api/schema#inclusive

**Comment by philippkuehn** [comment]
Yes, this is a problem, but I don't know yet in which scope we want to solve this problem. Just adding keyboard shortcuts for this could be enough, but it also helps to [show this visually with another cursor](https://github.com/curvenote/prosemirror-codemark). That is rather hacky though.

**Comment by rfgamaral** [comment]
Keyboards shortcuts already work to "leave the mark", so it's not so bad as I originally thought.

It would be nice to have something like that for all marks, built-in, and less hacky. But I understand it might be tricky to implement.

Thanks for getting back to me on this, and hopefully you folks will be able to improve this in the future 👍

**Comment by bdbch** [comment]
I've reopened this issue and added it to our tracker. Thanks for pinging @rfgamaral

**Comment by bdbch** [comment]
bump

**Comment by chrislicodes** [comment]
Same issue here

**Comment by linuxhackr** [comment]
Any update on this?

**Comment by bdbch** [comment]
Sadly not. I didn't had time to get to this issue yet.

**Comment by olejorgenb** [comment]
I think I've seen other editors handle this (though, usually semi-buggy..) by using the right key to "break free" from the mark. I think this has only been active when there's nothing to the right of the mark[1]. Eg.: `BOLD|` pressing right would not move the cursor, but it would move "out" of the mark, disabling bold.

[1] which make sense, since if there's something there you can usually just move until you're outside the mark.

**Comment by C-Hess** [comment]
I'm currently experimenting with an interesting solution using zero-width characters:

```typescript
  addKeyboardShortcuts() {
    return {
      'Mod-e': () => this.editor.commands.toggleCode(),
      // right arrow
      'ArrowRight': () => this.editor.commands.command(({ tr }) => {
        tr.removeStoredMark(this.type)
        // Insert zero-width character to escape the mark
        tr.insertText('\u200B', tr.selection.from, tr.selection.to)
        return true;
      }),
    }
  },
```

An advantage of this approach is we don't have to "simulate" a separate cursor. Now the tougher question is how can we strip out this zero-width character. Maybe I could create some pseudo-node that removes itself whenever content is added to it and insert that instead... I'll try and mess around with this more when I have the time

**Comment by habibayman** [comment]
I'd like to work with this can you assign me? @bdbch

**Comment by bdbch** [comment]
Feel free to look into it @habibayman

**Comment by habibayman** [comment]
Can we talk about the exact desired behavior? I believe every node has like a get away from this so why are we still calling this an issue, what are we seeking? @bdbch

**Comment by bdbch** [comment]
@rfgamaral is this still a bug to you? seems like @habibayman wasn't able to reproduce it and I can't remember getting stuck in code lines as well.

**Comment by Nantris** [comment]
Am I wrong that this isn't a bug - this is just how ProseMirror works?? You can set the mark to non-inclusive if you don't want this behavior. I don't think expecting TipTap to account for every possible preference in how things work is reasonable.

**Comment by habibayman** [comment]
@Nantris yeah I think this issue should be closed tbh
